### PR TITLE
Use primary constructors in features layer

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This approach differs from the last in that we preserve redundant fields.  We can consider followup PRs that then remove these.   It does make the code somewhat more confusing to keep these in, as these state values are themselves never read from other instances.  But this may smooth out the transition to using these.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal.

File: eng/Versions.props [hunk 1]
```diff
@@ -15,7 +15,7 @@
       Keep the setting conditional. The toolset sets the assembly version to 42.42.42.42 if not set explicitly.
     -->
     <AssemblyVersion Condition="'$(OfficialBuild)' == 'true' or '$(DotNetUseShippingVersions)' == 'true'">$(MajorVersion).$(MinorVersion).0.0</AssemblyVersion>
-    <MicrosoftNetCompilersToolsetVersion>4.7.0-1.final</MicrosoftNetCompilersToolsetVersion>
+    <MicrosoftNetCompilersToolsetVersion>4.7.0-3.23329.1</MicrosoftNetCompilersToolsetVersion>
```
Comment (CyrusNajmabadi) [inline eng/Versions.props:18]: @333fred ptal.

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Analyzers/Core/CodeFixes/AddParameter/CodeFixData.cs [hunk 2]
```diff
@@ -8,31 +8,25 @@
...
-            Method = method ?? throw new ArgumentNullException(nameof(method));
-            CreateChangedSolutionNonCascading = createChangedSolutionNonCascading ?? throw new ArgumentNullException(nameof(createChangedSolutionNonCascading));
-            CreateChangedSolutionCascading = createChangedSolutionCascading;
-        }
 
         /// <summary>
         /// The overload to fix.
         /// </summary>
-        public IMethodSymbol Method { get; }
+        public IMethodSymbol Method { get; } = method ?? throw new ArgumentNullException(nameof(method));
```
Comment (sharwell) [inline src/Analyzers/Core/CodeFixes/AddParameter/CodeFixData.cs:20]: ❔ What does the default constructor look like for a struct with a primary constructor and field initializers?

Comment (CyrusNajmabadi) [reply src/Analyzers/Core/CodeFixes/AddParameter/CodeFixData.cs:20]: the same as when you have a normal struct with a normal constructor that assigns to the fields.

File: src/Features/Core/Portable/CodeLens/ReferenceMethodDescriptor.cs [hunk 3]
```diff
@@ -12,40 +12,24 @@ namespace Microsoft.CodeAnalysis.CodeLens
...
-
-        /// <summary>
-        /// Describe a caller method of a callee
-        /// </summary>
-        /// <param name="fullName">Method's fully qualified name</param>
-        /// <param name="filePath">Method full path</param>
-        /// <remarks>
-        ///  Method full name is expected to be in the .NET full name type convention. That is,
-        ///  namespace/type is delimited by '.' and nested type is delimited by '+'
-        /// </remarks>
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/CodeLens/ReferenceMethodDescriptor.cs:43]: Fixed the fixer, and resent the PR with the doc comments properly presrved..

**Pull request merged**